src/components/common/PeopleTable.tsx

```diff
@@ -0,0 +1,260 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import Button from '@material-ui/core/Button';
+import makeStyles from '@material-ui/styles/makeStyles';
+import MaterialTable, { Query, Options, Column } from 'material-table';
+import React, { useState, useEffect } from 'react';
+
+import { ActionButtonContainer } from 'components/common/ActionButtonContainer';
+import { tableIcons } from 'components/common/TableIcons';
+import { BasicUserDetails, UserRole } from 'generated/sdk';
+import { useDataApi } from 'hooks/common/useDataApi';
+
+function sendUserRequest(
+  searchQuery: Query<any>,
+  api: any,
+  setLoading: React.Dispatch<React.SetStateAction<boolean>>,
+  selectedUsers: number[] | undefined,
+  selectedParticipants: number[],
+  userRole: UserRole | null
+) {
+  const variables = {
+    filter: searchQuery.search,
+    offset: searchQuery.pageSize * searchQuery.page,
+    first: searchQuery.pageSize,
+    subtractUsers: selectedUsers ? selectedUsers : [],
+    userRole: userRole,
+  };
+
+  setLoading(true);
+
+  return api()
+    .getUsers(variables)
+    .then((data: any) => {
+      setLoading(false);
+
+      return {
+        page: searchQuery.page,
+        totalCount: data?.users?.totalCount,
+        data: data?.users?.users.map((user: BasicUserDetails) => {
+          return {
+            firstname: user.firstname,
+            lastname: user.lastname,
+            organisation: user.organisation,
+            id: user.id,
+            tableData: { checked: selectedParticipants.includes(user.id) },
+          };
+        }),
+      };
+    });
+}
+
+type PeopleTableProps<T extends BasicUserDetails = BasicUserDetails> = {
+  selection: boolean;
+  isLoading?: boolean;
+  title?: string;
+  action?: {
+    fn: (data: any) => void;
+    actionIcon: JSX.Element;
+    actionText: string;
+  };
+  isFreeAction?: boolean;
+  data?: T[];
+  search?: boolean;
+  onRemove?: (item: any) => void;
+  onUpdate?: (item: any[]) => void;
+  selectedUsers?: number[];
+  mtOptions?: Options;
+  columns?: Column<any>[];
+  userRole?: UserRole;
+};
+
+const useStyles = makeStyles({
+  tableWrapper: {
+    '& .MuiToolbar-gutters': {
+      paddingLeft: '0',
+    },
+  },
+  verticalCentered: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+});
+
+const columns = [
+  { title: 'Name', field: 'firstname' },
+  { title: 'Surname', field: 'lastname' },
+  { title: 'Organisation', field: 'organisation' },
+];
+
+const PeopleTable: React.FC<PeopleTableProps> = props => {
+  const { isLoading } = props;
```

Comment:
It would be better to use default initializer set to `false` and remove `?? false` in the state hook